File: 20161103-java-racing_car_katas_2/src/main/java/tddmicroexercises/tirepressuremonitoringsystem/Alarm.java
package tddmicroexercises.tirepressuremonitoringsystem;

public class Alarm {
    private final double LowPressureTreshold = 17;
    private final double HighPressureTreshold = 21;

    private Sensor sensor;
    private boolean alarmOn = false;

    public Alarm(Sensor sensor) {
        this.sensor = sensor;
    }

    public void check() {
        double psiPressureValue = sensor.popNextPressurePsiValue();

        if (psiPressureValue < LowPressureTreshold || HighPressureTreshold < psiPressureValue) {
            alarmOn = true;
        }
    }

    public boolean isAlarmOn() {
        return alarmOn;
    }
}

File: 20170504-csharp-tire_pressure_monitoring_system/tire_pressure_monitoring_system/Alarm.cs
using System;
namespace tire_pressure_monitoring_system
{
	public class Alarm
	{
		private const double LowPressureTreshold = 17;
		private const double HighPressureTreshold = 21;

		ISensor _sensor;

		bool _alarmOn = false;
		private long _alarmCount = 0;

        public Alarm(ISensor sensor) {
            _sensor = sensor;
        }

		public void Check()
		{
			double psiPressureValue = _sensor.PopNextPressurePsiValue();

			if (psiPressureValue < LowPressureTreshold || HighPressureTreshold < psiPressureValue)
			{
				_alarmOn = true;
				_alarmCount += 1;
			}
		}

		public bool AlarmOn
		{
			get { return _alarmOn; }
		}

	}
}
